src/status_im2/common/nickname_drawer/view.cljs

```diff
@@ -0,0 +1,84 @@
+(ns status-im2.common.nickname-drawer.view
+  (:require [clojure.string :as string]
+            [quo.design-system.colors :as colors]
+            [quo2.components.icon :as icons]
+            [quo2.core :as quo]
+            [react-native.core :as rn]
+            [reagent.core :as reagent]
+            [status-im2.common.nickname-drawer.style :as style]
+            [utils.i18n :as i18n]
+            [utils.re-frame :as rf]))
+
+(defn bad-nickname?
+  [nickname]
+  (string/blank? @nickname))
+
+(defn add-nickname-toast
+  [primary-name entered-nickname public-key]
```

Comment:
I think it would be better to use keys here.

Side note, I am starting to notice some inconsistency in our codebase for parameter definitions, so we should discuss and define some guidelines for when we should and shouldn't use keys as parameters.

Reply:
I think using maps is very useful when having a lot of parameters for a function or if we want to add default values .. But this function has only 3 parameters

Reply:
There are some people (like me) who prefer to always use maps for component arguments, such that components are like React components where the argument is always a single map of props. This convention of course only holds for components, not for other functions, where defining such argument rules/conventions would be absurd in the Clojure community.